Add todo

robi1467 opened the pull request.

robi1467 committed:
Add test for addTodo

We added a server side failing test for adding a new Todo.

robi1467 committed:
Implement addTodo

We implemented addTodo--including validation-- in the TodoController. The server side implementation for adding a todo is done-- besides a little bit of testing.

robi1467 committed:
Create typescript for AddTodoComponent

It doesn't have an html template yet, or any routing or tests.
(Also, we still need to write the 'adding' part of the TodoService.)

robi1467 committed:
Add route to the Add Todos page

Including a mat-fab at the bottom of the todo-list component.

There is still not any html on this page, and the service isn't
implemented yet.

robi1467 committed:
Add stub addTodo method in todo service

robi1467 committed:
Add owner field in the add todo form

There isn't a submit button yet (or any testing), but the validation
is working.

robi1467 committed:
Add body and category into the html

We added a body the auto-resizes into the html and a input field for the category
We still need a submit button and proper tests.

robi1467 committed:
Added a button to submit the form

robi1467 committed:
Wrote addTodo in the TodoService

(Also, we added tests for this method, and we hooked up
a route in the server to handle the post request.)

helloworld12321 committed:
Add spec for AddTodoComponent

It has two (very small) tests inside it at the moment. Both are passing.

helloworld12321 committed:
Test the owner field

Mostly, make sure the validation is working.

These tests are all passing.

helloworld12321 committed:
Import MatIconModule into certain Karma tests

This fixes the warning "'mat-icon' is not a known element".

helloworld12321 committed:
Test the category and body fields

These tests—which are very similar to the user tests—are also currently passing.

helloworld12321 committed:
Test the status field

These tests already pass.

helloworld12321 committed:
Reformat todo.service.spec.ts

There was some indentation wonkiness (Again, an issue with VS Code).

helloworld12321 committed:
Test that AddTodoComponent makes status a boolean on the new Todo object

This test is currently failing, because AddTodoComponent makes status a string (either "complete" or "incomplete".)

helloworld12321 committed:
Make the previous test pass

AddTodoComponents now explicitly construct a Todo object rather than just assuming that addTodoForm.value conforms to the Todo interface (it doesn't).

helloworld12321 committed:
Add server-side tests that the validation works

These tests all pass already.

There isn't much server-side validation. Mostly, we're just checking that the JSON we get from the client is structured like a todo, and that none of the fields are empty.

robi1467 committed:
create add-todo.po

Created the foundation for the e2e test

robi1467 committed:
Fix routing error

In the add-todo there was an a routing error in the submitForm() method fixed it to wok for add-todo

robi1467 committed:
Add E2E test for addTodo

Add a test to test the actual addTodo button and a test where it adds a unique user than makes sure there is only one in all of the todos proving it was added

helloworld12321 committed:
Fix duplicate id in add-todo.component.html

This is just a simple copy-and-paste problem :)

helloworld12321 commented:
Hopefully that commit fixes LGTM's "Ambiguous HTML id attribute" warning 🙂 (I'm not too concerned about the unused imports!)

helloworld12321 commented:
Sorry for the force push—I wanted to reword that last commit message.

helloworld12321 commented:
Oops, still ambiguous!

helloworld12321 committed:
Fix other duplicate id in add-todo.component.html

There were three elements named #owner-error in add-todo.component.html. Commit e5b0d9b fixed one of the duplicates. This commit fixes the other.

helloworld12321 commented:
There we go! 😊
